[SDK-3027] Update swift analytics section (#4823)
braze-inc/braze-docs: _docs/_developer_guide/platform_integration_guides/swift/analytics/setting_custom_attributes.md

The following is an example of setting the first name attribute:

{% tabs %}
{% tab swift %}

```swift
AppDelegate.braze?.user.set(firstName: "first_name")
```

{% endtab %}
{% tab OBJECTIVE-C %}

```objc

> AppDelegate.braze.user.firstName = @"first_name";

Reviewer: ```suggestion
[AppDelegate.braze.user setFirstName:@"first_name"];
```

The `AppDelegate.braze.user.firstName = @"first_name";` syntax is deprecated

{% tab OBJECTIVE-C %}

```objc

> AppDelegate.braze.user.firstName = @"first_name";

Reviewer: ```suggestion
[AppDelegate.braze.user setFirstName:@"first_name"];
```

```

{% endtab %}
{% tab swift %}

```swift

> AppDelegate.braze?.user.firstName = "first_name"

Reviewer: ```suggestion
AppDelegate.braze?.user.set(firstName: "first_name")
```

```

{% endtab %}
{% endtabs %}

> The following attributes should be set on the `Braze.User` object:

Reviewer: Can we remove `userID` from the list? We don't provide an API to modify it directly. It's always modified via `changeUser()`

{% tab OBJECTIVE-C %}

```objc

> [AppDelegate.braze.user setCustomAttributeWithKey:@"your_attribute_key" andDateValue:yourDateValue];

Reviewer: Can we remove the paragraph above

> Dates passed to Braze with this method must either be in the [ISO 8601][2] format (e.g `2013-07-16T19:20:30+01:00`) or in the `yyyy-MM-dd'T'HH:mm:ss:SSSZ` format (`2016-12-14T13:32:31.601-0800`).

Our API supports the Date type directly now.

#### Additional information

- Refer to the [`Braze.User` documentation][6] for more information.

## Setting up user subscriptions

To set up a subscription for your users (either email or push), call the functions `set(emailSubscriptionState:)` or `set(pushNotificationSubscriptionState:)`, respectively. Both of these functions take the enum type `Braze.User.SubscriptionState` as arguments. This type has three different states:

| Subscription Status | Definition |
| ------------------- | ---------- |
| `optedIn` | Subscribed, and explicitly opted in |
| `subscribed` | Subscribed, but not explicitly opted in |
| `unsubscribed` | Unsubscribed and/or explicitly opted out |
{: .reset-td-br-1 .reset-td-br-2}

Users who grant permission for an app to send them push notifications default to the status of `optedIn` as iOS requires an explicit opt-in.

> Users will be set to `subscribed` automatically upon receipt of a valid email address; however, we suggest that you establish an explicit opt-in process and set this value to `OptedIn` upon receipt of explicit consent from your user. Refer to [Managing user subscriptions]({{site.baseurl}}/user_guide/message_building_by_channel/email/managing_user_subscriptions/) for more details.

Reviewer: ```suggestion
Users will be set to `subscribed` automatically upon receipt of a valid email address; however, we suggest that you establish an explicit opt-in process and set this value to `optedIn` upon receipt of explicit consent from your user. Refer to [Managing user subscriptions]({{site.baseurl}}/user_guide/message_building_by_channel/email/managing_user_subscriptions/) for more details.
```

#### Additional information

- Refer to the [`Braze.User` documentation][6] for more information.

## Setting up user subscriptions

> To set up a subscription for your users (either email or push), call the functions `setEmailNotificationSubscriptionType` or `setPushNotificationSubscriptionType`, respectively. Both of these functions take the enum type `Braze.User.SubscriptionState` as arguments. This type has three different states:

Reviewer: ```suggestion
To set up a subscription for your users (either email or push), call the functions `set(emailSubscriptionState:)` or `set(pushNotificationSubscriptionState:)`, respectively. Both of these functions take the enum type `Braze.User.SubscriptionState` as arguments. This type has three different states:
```

| Subscription Status | Definition |
| ------------------- | ---------- |
| `optedIn` | Subscribed, and explicitly opted in |
| `subscribed` | Subscribed, but not explicitly opted in |
| `unsubscribed` | Unsubscribed and/or explicitly opted out |
{: .reset-td-br-1 .reset-td-br-2}

Users who grant permission for an app to send them push notifications default to the status of `optedIn` as iOS requires an explicit opt-in.

Users will be set to `subscribed` automatically upon receipt of a valid email address; however, we suggest that you establish an explicit opt-in process and set this value to `OptedIn` upon receipt of explicit consent from your user. Refer to [Managing user subscriptions]({{site.baseurl}}/user_guide/message_building_by_channel/email/managing_user_subscriptions/) for more details.

### Setting email subscriptions

{% tabs %}
{% tab OBJECTIVE-C %}

```objc

> [AppDelegate.braze.user setEmailNotificationSubscriptionType: Braze.User.SubscriptionState]

Reviewer: ```suggestion
[AppDelegate.braze.user setEmailSubscriptionState: BRZUserSubscriptionState]
```

{% tab OBJECTIVE-C %}

```objc

> [AppDelegate.braze.user setPushNotificationSubscriptionType: Braze.User.SubscriptionState]

Reviewer: ```suggestion
[AppDelegate.braze.user setPushNotificationSubscriptionState: BRZUserSubscriptionState]
```
